src/pattern.jl

```diff
@@ -389,3 +389,9 @@ function hessian_pattern_to_mat(
 ) where {P1,P2,T<:HessianTracer,D1<:Dual{P1,T},D2<:Dual{P2,T}}
     return hessian_pattern_to_mat(tracer.(xt), tracer(yt))
 end
+
+function hessian_pattern_to_mat(
+    xt::AbstractArray{D1}, yt::Number
+) where {P1,T<:HessianTracer,D1<:Dual{P1,T}}
+    return hessian_pattern_to_mat(tracer.(xt), empty(T))
+end
```

Comment:
Might just as well return an empty matrix here instead of calling `hessian_pattern_to_mat`.